src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package taxi.service;
+
+import java.util.Optional;
+import javax.naming.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.get().getPassword().equals(password) && !driver.isEmpty()) {
```

Comment:
```suggestion
        if ( !driver.isEmpty() && driver.get().getPassword().equals(password)) {
```
you will have an exception in case driver's optional has null if you will be using your version